lib/web/desktop_playback.go

```diff
@@ -38,6 +38,7 @@ func (h *Handler) desktopPlaybackHandle(
 	p httprouter.Params,
 	sctx *SessionContext,
 	site reversetunnelclient.RemoteSite,
+	ws *websocket.Conn,
 ) (interface{}, error) {
 	sID := p.ByName("sid")
 	if sID == "" {
```

Comment:
This needs to be cleaned up a bit.

Prior to your change, the connection had not been upgraded to a websocket here, so it was perfectly fine to return an error (the middleware would pass the error along in the HTTP response).

Now that the connection is upgraded to a websocket, returning an error here will cause issues, as we can't write a plain HTTP response.

I wonder if there's a way to make this harder to mess up. Perhaps `WithClusterAuthWebsocket` should accept a func with a different signature so that it's impossible to return `interface{}, error`.

Reply:
`WithClusterAuthWebSocket` will write the error from the handler to the websocket, does that work okay? it seems in this function it already does something similar with errors further down

Reply:
I think we're actually OK here. I tried to inject errors at different points before and after these changes and they all showed up in the UI in the same way.

Sorry for the false alarm.